src/canopen_comm.cpp

```diff
@@ -0,0 +1,191 @@
+        {send_runtime_query::READ_SPEKTRUM_RADIO_CAPTURE, {0x211a, 2}}, 
+        {send_runtime_query::READ_DESTINATION_POSITION_REACHED_FLAG, {0x211b, 3}}, 
+        {send_runtime_query::READ_MEMS_ACCELEROMETER_AXIS, {0x211c, 0}}, 
+        {send_runtime_query::READ_MAGSENSOR_TRACK_DETECT, {0x211d, 3}}, 
+        {send_runtime_query::READ_MAGSENSOR_TRACK_POSITION, {0x211e, 2}},
+        {send_runtime_query::READ_MAGSENSOR_MARKERS, {0x211f, 3}}, 
+        {send_runtime_query::READ_MAGSENSOR_STATUS, {0x2120, 2}}, 
+        {send_runtime_query::READ_MOTOR_STATUS_FLAGS, {0x2121, 2}}, 
+    };
+
+    canopen_comm::canopen_comm(canid_t roboteq_can_id, std::string ifname): _roboteq_can_id(roboteq_can_id) {
+
+        if((_socket_handle = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+            printf("Error while opening socket");
+            throw -1;
+        }
+
+        struct ifreq ifr;
+
+        std::strcpy(ifr.ifr_name, ifname.c_str());
+        ioctl(_socket_handle, SIOCGIFINDEX, &ifr);
+
+        struct sockaddr_can addr;
+        
+        addr.can_family  = AF_CAN;
+        addr.can_ifindex = ifr.ifr_ifindex;
+
+        std::cout << ifname << " at index " << ifr.ifr_ifindex << std::endl;
+
+        struct can_filter rfilter[2];
+
+        rfilter[0].can_id   = _sdo_cob_id_offset + roboteq::canopen_comm::_roboteq_can_id;
+        rfilter[0].can_mask = CAN_SFF_MASK;
+        rfilter[1].can_id   = _sdo_response_cob_id_offset + roboteq::canopen_comm::_roboteq_can_id;
+        rfilter[1].can_mask = CAN_SFF_MASK;
+
+        setsockopt(_socket_handle, SOL_CAN_RAW, CAN_RAW_FILTER, &rfilter, sizeof(rfilter));
+
+        if(bind(_socket_handle, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
+            printf("Error in socket bind");
+            throw -2;
+        }
+    
+    }
+
+    bool canopen_comm::sdo_download(send_runtime_command command, uint8_t subindex, uint32_t data) {
+
+        struct can_frame frame;
+
+        frame.can_id  = _sdo_cob_id_offset + roboteq::canopen_comm::_roboteq_can_id;
+        frame.can_dlc = 8;
+        frame.data[0] = (_sdo_command<<4) | (canopen_comm::_runtime_command_map.at(command).number_of_unused_bytes<<2);
+        frame.data[1] = canopen_comm::_runtime_command_map.at(command).canopen_index; 
+        frame.data[2] = canopen_comm::_runtime_command_map.at(command).canopen_index >> 8;
+        frame.data[3] = subindex;
+        frame.data[4] = data >> 0;
+        frame.data[5] = data >> 8;
+        frame.data[6] = data >> 16;
+        frame.data[7] = data >> 24;
+
+        write(roboteq::canopen_comm::_socket_handle, &frame, sizeof(struct can_frame));
+
+        struct can_frame response_frame = {};
+
+        uint32_t nbytes = read(roboteq::canopen_comm::_socket_handle, &response_frame, sizeof(struct can_frame));
+
+        std::cout <<  std::hex << response_frame.can_id << "\t" << static_cast<unsigned>(response_frame.can_dlc) << "\t" << static_cast<unsigned>(response_frame.data[0])
```

Comment:
For now, comment out ALL print statements. Since this driver could be used in non-ROS applications, we should a way to log both to rosconsole or to std console. Here's a good example on how to do it:
https://answers.ros.org/question/29874/putting-log-messages-from-non-ros-code-into-ros-logging-system/ (except u should use the native function pointers)

This can be done in a later pr